kubejs/server_scripts/enigmatica/kubejs/base/recipes/shapeless.js

```diff
@@ -310,6 +322,30 @@ events.listen('recipes', (event) => {
         event.shapeless(`powah:reactor_${tier}`, ['#powah:reactor', capacitor]);
     });
 
+    colors.forEach(function (color) {
+        let otherColors = [],
+            i = 0,
+            j = colors.indexOf(color);
+
+        while (i < j) {
+            otherColors.push(colors[i]);
+            i++;
+        }
+
+        let otherSimplePots = otherColors.map((otherColor) => `botanypots:${otherColor}_botany_pot`),
+            otherHopperPots = otherColors.map((otherColor) => `botanypots:hopper_${otherColor}_botany_pot`);
+
+        otherSimplePots.push('botanypots:botany_pot');
+        otherHopperPots.push('botanypots:hopper_botany_pot');
+
+        event.shapeless(`botanypots:${color}_botany_pot`, [Ingredient.of(otherSimplePots), `#forge:dyes/${color}`]);
+
+        event.shapeless(`botanypots:hopper_${color}_botany_pot`, [
+            Ingredient.of(otherHopperPots),
+            `#forge:dyes/${color}`
+        ]);
+    });
+

```

Comment:
I don't really understand what's going on here - Is the extra looping and logic to prevent crafting a pot into the same pot?

Reply:
Yep. Though now that you mention it I think I can do this with an array filter instead of the loop.

Reply:
I don't see the point to be honest :P

Reply:
Mostly just to keep things tidy in game